src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1

```diff
@@ -1513,4 +1513,125 @@ class Build: ADOSVTBase
+            {
+                ## In case it is YAML build
+                if([Helpers]::CheckMember($this.ControlSettings,"Build.RegexForOAuthTokenInYAMLScript"))
+                {
+                    $orgName = $this.OrganizationContext.OrganizationName
+                    $projectName = $this.BuildObj.project.name
+                    $projectId = $this.BuildObj.project.id
+                    $buildId = $this.BuildObj.id
+                    $repoid = $this.BuildObj.repository.id
+                    $resultObj = @()
+                    $regex = $this.ControlSettings.Build.RegexForOAuthTokenInYAMLScript
+
+                    try{
+                        $url = "https://dev.azure.com/{0}/{1}/_apis/git/repositories/{2}/refs?api-version=6.0" -f $orgName,$projectName,$repoid
+                        $responseObj = @([WebRequestHelper]::InvokeGetWebRequest($url);)
+                        $branches = @($responseObj.name | Foreach-Object { $_.split("/")[-1]})
+                        if($branches.count -gt 0)
+                        {
+                            $branches | where-object {
+                                $currentBranch = $_    
+                                $refobj = "" | Select-Object branch,fileName
+                                $refobj.branch = $currentBranch
+                                try{
+                                    $url = 'https://dev.azure.com/{0}/{1}/_apps/hub/ms.vss-build-web.ci-designer-hub?pipelineId={2}&branch={3}&__rt=fps&__ver=2' -f $orgName, $projectId , $buildId, $currentBranch;
+                                    $responseObj = @([WebRequestHelper]::InvokeGetWebRequest($url);)
+                                    if([Helpers]::CheckMember($responseObj,"fps.dataProviders.data") -and $responseObj.fps.dataProviders.data.'ms.vss-build-web.pipeline-editor-data-provider' -and [Helpers]::CheckMember($responseObj.fps.dataProviders.data.'ms.vss-build-web.pipeline-editor-data-provider',"content") -and  $responseObj.fps.dataProviders.data.'ms.vss-build-web.pipeline-editor-data-provider'.content)
+                                    {
+                                        $dataprovider = $responseObj.fps.dataProviders.data.'ms.vss-build-web.pipeline-editor-data-provider'
+                                        $yamlFileContent = $dataprovider.content
+                                        if($yamlFileContent -match $regex)
+                                        {
+                                            $refobj.fileName = $dataprovider.definition.process.yamlFilename
+                                            $resultObj += $refobj
+                                        }
+                                    }
+                                }
+                                catch
+                                {
+                                    $controlResult.AddMessage([VerificationResult]::Error,"Not able to fetch YAML file for the branch: $($currentBranch)");
```

Comment:
You should eat exception for this catch block, because even if there is any error then also control will pass in next else statement.

Reply:
added another check to resolve this issue